client_test.go

```diff
@@ -165,22 +165,18 @@ func TestCaptureException(t *testing.T) {
 					Type:  "*sentry.customErr",
 					Value: "wat",
 					Mechanism: &Mechanism{
-						Data: map[string]any{
-							"is_exception_group": true,
-							"exception_id":       0,
-						},
+						ExceptionID:      0,
+						IsExceptionGroup: true,
 					},
 				},
 				{
 					Type:       "*errors.withStack",
 					Value:      "wat",
 					Stacktrace: &Stacktrace{Frames: []Frame{}},
 					Mechanism: &Mechanism{
-						Data: map[string]any{
-							"exception_id":       1,
-							"is_exception_group": true,
-							"parent_id":          0,
-						},
+						ExceptionID:      1,
+						ParentID:         Pointer(0),
```

Comment:
not a fan of the pointer type here. It's not referring to a memory address

Reply:
The reason for using pointer is that for first exception, it expects `0` for exception_id and nothing for parent_id. The second exception is `{"exception_id":1,"parent_id":0}`. Since go defaults to 0 for zero int values, it wouldn't work for first exception.

It's either using pointers (simpler) or a nullable type with custom marshal function (something like sql.NullInt16).